go/vt/vtgate/engine/update.go

```diff
@@ -218,3 +224,21 @@ func (upd *Update) updateVindexEntries(vcursor VCursor, query string, bindVars m
 	}
 	return nil
 }
+
+func (upd *Update) execUpdateByDestination(vcursor VCursor, bindVars map[string]*querypb.BindVariable, dest key.Destination) (*sqltypes.Result, error) {
+	rss, _, err := vcursor.ResolveDestinations(upd.Keyspace.Name, nil, []key.Destination{dest})
+	if err != nil {
+		return nil, vterrors.Wrap(err, "execDeleteScatter")
+	}
+
+	queries := make([]*querypb.BoundQuery, len(rss))
+	sql := sqlannotation.AnnotateIfDML(upd.Query, nil)
+	for i := range rss {
+		queries[i] = &querypb.BoundQuery{
+			Sql:           sql,
+			BindVariables: bindVars,
+		}
+	}
+	autocommit := (len(rss) == 1) && vcursor.AutocommitApproval()
```

Comment:
For completeness I think we should include the MultiShardAutocommit support in this PR as well or at the very least do so right after merging.